docs/addons/calico.md

```diff
@@ -0,0 +1,190 @@
+Follow the instructions found [here](https://docs.projectcalico.org/getting-started/clis/calicoctl/install), under *Install calicoctl as a binary on a single host* it will include instructions for your specific OS. You can test that calicoctl was installed correctly using the following:
+
+```bash
+calicoctl version
+```
+
+You should see output similar to the following:
+
+```
+Client Version:    v3.19.1
+Git commit:        6fc0db96
+Cluster Version:   v3.15.1
+Cluster Type:      typha,kdd,k8s,ecs
+```
+
+### Pod to Pod communications with no policies
+
+If you deploy [App of Apps installed using ArgoCD](https://github.com/aws-quickstart/quickstart-ssp-amazon-eks/blob/feature/calico/docs/getting-started.md#deploy-workloads-with-argocd), you will notice that there are no network policies. 
+
+To verify this, run the following:
+
+```bash
+kubectl get networkpolicy -A
+```
+You should see the following output:
+
+```
+No resource found
+```
+
+This means that any resources within the cluster should be able to make ingress and egress connections with other resources within and outside the cluster. You can verify, for example, that you are able to ping from `team-riker` pod to `team-burnham` pod.
+
+First we retrieve the pod name from the `team-burnham` namespace its podIP:
+
+```bash
+BURNHAM_POD=$(kubectl get pod -n team-burnham -o jsonpath='{.items[0].metadata.name}') 
+BURNHAM_POD_IP=$(kubectl get pod -n team-burnham $BURNHAM_POD -o jsonpath='{.status.podIP}')
+BURNHAM_POD_DEP=$(kubectl get deployments -n team-burnham -o jsonpath='{.items[0].metadata.name}')
+```
+
+Now we can start a shell from the pod in the `team-riker` namespace and ping the pod from `team-burnham` namespace:
+
+```bash
+RIKER_POD=$(kubectl -n team-riker get pod -o jsonpath='{.items[0].metadata.name}')
+kubectl exec --stdin --tty -n team-riker $RIKER_POD -- /bin/bash
+```
+
+Note: since this opens a shell inside the pod, it will not have the environment variables saved above. You should retrieve the actual podIP and the deployment name from the environment variables `BURNHAM_POD` and `BURNHAM_POD_DEP`.
+
+With those actual values, ping the DNS of the pod from `team-burnham`:
+
+```bash
+root@kustomize-guestbook-ui-85985d774c-tvk6n:/var/www/html# ping <podIP>.<deployment name>.team-burnham.svc.cluster.local
+```
+
+You should see it pinging successfully:
+
+```
+PING 10-0-184-139.guestbook-ui.team-burnham.svc.cluster.local (10.0.184.139): 56 data bytes
+64 bytes from 10.0.184.139: icmp_seq=0 ttl=254 time=0.101 ms
+64 bytes from 10.0.184.139: icmp_seq=1 ttl=254 time=0.069 ms
+64 bytes from 10.0.184.139: icmp_seq=2 ttl=254 time=0.069 ms
+64 bytes from 10.0.184.139: icmp_seq=3 ttl=254 time=0.073 ms
+```
+### Applying Policies using Calico to block traffic
+
+One of the powerful things about Calico is the ability to apply a Global Network Policy - policy across all namespaces.
+
+Let's apply the following Global Network Policy:
+
+```yaml
+apiVersion: projectcalico.org/v3
```

Comment:
This requires CRD, correct? We may need an example with vanilla network policy.

Reply:
Are we going to show both examples then? Currently there are two CRDs shown in the example:
- One that blocks all the resource communication other than kube-system ns
- One that will allow ping connection from team-riker to team-burnham

I can probably write one of them to use a vanilla network policy. I'm just a bit confused as the idea of vanilla network policy was going to be addressed in the other Issue (#31 )

Reply:
The reason why we use Calico is not because we want to promote their specific features, but just for regular network policy support which is not available with vpc-cni. Atm, if customer does not use Calico addon, there will be no network policy support at any level. 

#31 is addressing example  in code for our SSP patterns repository which will also have documentation in the SSP patterns describing it.

For this documentation task it makes sense to call out that generic network policy can also be enabled with any other alternate CNI providers (which we have none atm) while global network policy binds customers to Calico.

Reply:
Understood. I've included doc changes to call out the distinction between k8s network policy & CRDs and replaced CRDs with k8s in the example. Also included the reason we use Calico